pure.zsh

```diff
@@ -403,7 +403,6 @@ prompt_pure_async_tasks() {
 		unset prompt_pure_git_dirty
 		unset prompt_pure_git_last_dirty_check_timestamp
 		unset prompt_pure_git_arrows
-		unset prompt_pure_git_stash
```

Comment:
Not related to this fix, this variable was unused, but overlooked and sneaked in with #462. 

I should have at least mentioned that in the commit message, or better yet, have created a separate commit for this cleanup.